kotlinx-coroutines-core/common/src/Await.kt

```diff
@@ -7,6 +7,14 @@ package kotlinx.coroutines
 import kotlinx.atomicfu.*
 import kotlin.coroutines.*
 
+/**
+ * Suspends until cancellation, in which case it will throw a [CancellationException].
+ *
+ * Handy because it returns [Nothing], allowing it to be used in any coroutine,
+ * regardless of the required return type.
+ */
+public suspend inline fun awaitCancellation(): Nothing = suspendCancellableCoroutine {}
```

Comment:
It seems more efficient in terms of the resulting bytecode to make it not-inline. Note, that you'll need to update API dump after that. See https://github.com/Kotlin/kotlinx.coroutines/blob/master/CONTRIBUTING.md#updating-the-public-api-dump

Also, it is very hard to pick the "right" file to place this declaration in, but I think it might be better to add to `Delay.kt` and actually mention it in the docs of the `delay` function. Since the current workaround is to write `delay(Long.MAX_VALUE)`, it could say that this is equivalent to `awaitCancellation()`.

Reply:
Done. BTW, thanks for updating the contributing guide.